app/validation/validator.py

```diff
@@ -491,38 +487,38 @@ def _validate_answer_ids_present_in_schema(self, when, answer_ids_with_group_id,
         :return: list of dictionaries containing error messages, otherwise it returns an empty list
         """
         errors = []
-        answer_reference_id_fields = ('id', 'answer_ids', 'comparison_id')
+        answer_reference_id_fields = ('id', 'comparison_id')
         present_id_ref_keys = [x for x in answer_reference_id_fields if x in when]
 
         for id_ref_key in present_id_ref_keys:
-            # answer_ids is a list, so we have to handle it a little differently to get the same error message.
-            if id_ref_key == 'answer_ids':
-                for answer_id in when['answer_ids']:
-                    if answer_id not in answer_ids_with_group_id:
-                        errors.append(self._error_message('The answer id - {} in the {} key of the "when" clause for {} does not exist'
-                                                          .format(answer_id, id_ref_key, referenced_id)))
-            else:
-                if when[id_ref_key] not in answer_ids_with_group_id:
-                    errors.append(self._error_message('The answer id - {} in the {} key of the "when" clause for {} does not exist'
-                                                      .format(when[id_ref_key], id_ref_key, referenced_id)))
+            if when[id_ref_key] not in answer_ids_with_group_id:
+                errors.append(self._error_message('The answer id - {} in the {} key of the "when" clause for {} does not exist'
+                                                  .format(when[id_ref_key], id_ref_key, referenced_id)))
 
         return errors
 
-    def _validate_type_key_in_when_rule(self, when, referenced_id):
+    def _validate_condition_in_when_rule(self, when, answer_ids_with_group_id):
         """
         Validate when rule with 'type' key is correct
         :return: list of dictionaries containing error messages, otherwise it returns an empty list
         """
         errors = []
 
-        if when['type'] == 'answer_count' and 'answer_ids' not in when:
-            errors.append(self._error_message('"answer_ids" key has to be included when type is "answer_count" in a "when" clause'))
-        else:  # We know we have a 'answer_ids' key so can do validation against it
-            if when['condition'] in ('contains', 'not contains'):
-                errors.append(self._error_message('The condition "{}" is not valid for an answer_count based "when" clause'
-                                                  .format(when['condition'])))
-            if len(when['answer_ids']) != len(set(when['answer_ids'])):
-                errors.append(self._error_message('Duplicate answer ids found within {} clause'.format(referenced_id)))
+        condition = when['condition']
+        checkbox_exclusive_conditions = ('contains any', 'contains all')
+        invalid_checkbox_conditions = ('equals any', 'not equals any')
+        conditions_requiring_list_match_values = (*invalid_checkbox_conditions, *checkbox_exclusive_conditions)
```

Comment:
The validation for checkbox conditions and conditions requiring list match values should be more clearly separated. I think making `conditions_requiring_list_match_values` an explicit list (rather than trying to make use of the checkbox conditions) and moving it to after the checkbox validation is done will help.

Reply:
Fair, done 👍